### ISSUE TITLE
Expand the `sunpy.coordinates` test for complex cases

### ISSUE THREAD

**Issue by ayshih** [sunpy/sunpy#7893]
### Provide a general description of the issue or problem.

We now have a dedicated CI build for `sunpy.coordinates`. The goal now is to create specific tests which cover harder edge cases, provide more coverage of coordinate transforms etc.

- Running online tests – namely ones that involve downloading a JPL ephemeris – without having to run online tests under `sunpy.net`
- More expansive `hypothesis`-based testing without affecting all other CI builds
- Maybe some other benefits

**Comment by Diya910** [comment]
I’m working on the feature and need guidance on testing the CI build functionality. Could you please advise on the best way to validate if the CI build is working correctly and any specific tests or tools I should use?

**Comment by nabobalis** [comment]
You can run the new tox test environment locally: https://tox.wiki/en/4.23.2/user_guide.html 

For checking it works on GitHub actions, you can enable github actions on your own fork and test it that way before you do a pull request. When you open the pull request to sunpy, it will also run there if its configured.

**Comment by parharti** [comment]
is this issue still open?

**Comment by nabobalis** [comment]
Yes but the issue is now to create the specific tests required. This is a complex task and I do not recommend anyone who does not understand what is needed to complete this issue to work on it.

**Comment by parharti** [comment]
got it
